test/integration/etcd/server.go

```diff
@@ -152,19 +138,24 @@ func StartRealAPIServerOrDie(t *testing.T, configFuncs ...func(*options.ServerRu
 
 	kubeClient := clientset.NewForConfigOrDie(kubeClientConfig)
 
+	stopCh := make(chan struct{})
+	errCh := make(chan error)
 	go func() {
 		// Catch panics that occur in this go routine so we get a comprehensible failure
 		defer func() {
 			if err := recover(); err != nil {
 				t.Errorf("Unexpected panic trying to start API server: %#v", err)
 			}
 		}()
+		defer close(errCh)
 
 		prepared, err := kubeAPIServer.PrepareRun()
 		if err != nil {
-			t.Error(err)
+			errCh <- err
+			return
```

Comment:
in `cmd/kube-apiserver/app/testing/testserver.go` the ` wait.PollImmediate` that waits for the /healthz endpoint checks this channel
```go
		select {
		case err := <-errCh:
			return false, err
		default:
		}
```

if PrepareRun doesn't work we can fail faster this way, otherwise I assume we'll fail after waiting for healthz to be ok

Reply:
good point - added.